zipkin-collector/rabbitmq/src/main/java/zipkin/collector/rabbitmq/RabbitMqCollector.java

```diff
@@ -0,0 +1,213 @@
+/**
+ * Copyright 2015-2017 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin.collector.rabbitmq;
+
+import com.rabbitmq.client.Address;
+import com.rabbitmq.client.Connection;
+import com.rabbitmq.client.ConnectionFactory;
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import zipkin.collector.Collector;
+import zipkin.collector.CollectorComponent;
+import zipkin.collector.CollectorMetrics;
+import zipkin.collector.CollectorSampler;
+import zipkin.internal.LazyCloseable;
+import zipkin.storage.StorageComponent;
+
+import static zipkin.internal.Util.checkNotNull;
+
+/**
+ * This collector consumes encoded binary messages from a RabbitMQ queue.
+ */
+public final class RabbitMqCollector implements CollectorComponent {
+  private static final Logger LOG = LoggerFactory.getLogger(RabbitMqCollector.class);
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  /** Configuration including defaults needed to consume spans from a RabbitMQ queue. */
+  public static final class Builder implements CollectorComponent.Builder {
+    Collector.Builder delegate = Collector.builder(RabbitMqCollector.class);
+    CollectorMetrics metrics = CollectorMetrics.NOOP_METRICS;
+    String queue = "zipkin.spans";
```

Comment:
I'd prefer plain old zipkin here, as otherwise folks might accidentally suggest it in support and have that be a lie!